sentry-android-core/src/main/java/io/sentry/android/core/LifecycleWatcher.java

```diff
@@ -101,6 +108,29 @@ public void onBackground() {
     scheduleEndSession();
 
     addAppBreadcrumb("background");
+
+    if (enableLogFlushing) {
```

Comment:
hmm, wondering if it's a good idea to bind everything to LifecycleWatcher.. Since we now can addAppStateListener from anywhere, wdyt of introducing an `AndroidLoggerApi` which delegates everything to LoggerApi, but in addition subscribes for onBackground/onForeground to trigger logs flush? And then we override the Java LoggerApi via Options as always